File: src/main/java/managers/flag/FieldFlag.java
package managers.flag;

import util.packet.IntegerValue;

public enum FieldFlag implements IntegerValue {

    ID(0x01),
    RETURN_MAP(0x02),
    MAP_AREA(0x04),
    FOOTHOLDS(0x08),
    FORCED_RETURN(0x10),
    FIELD_LIMIT(0x20),
    NAME(0x40),
    ON_ENTER(0x80),
    PORTALS(0x100),
    AREAS(0x200),
    LIFE(0x400);

    private final int flag;

    FieldFlag(int flag) {
        this.flag = flag;
    }

    @Override
    public int getValue() {
        return flag;
    }

    @Override
    public void setValue(int value) {
        //...
    }
}

File: src/main/java/managers/flag/FieldFlag.kt
package managers.flag

import util.packet.IntegerValue

enum class FieldFlag(override val value: Int) : IntegerValue {
    ID(0x01),
    RETURN_MAP(0x02),
    MAP_AREA(0x04),
    FOOTHOLDS(0x08),
    FORCED_RETURN(0x10),
    FIELD_LIMIT(0x20),
    MAP_NAME(0x40),
    STREET_NAME(0x80),
    ON_ENTER(0x100),
    PORTALS(0x200),
    AREAS(0x400),
    LIFE(0x800),
    REACTOR(0x1000);
}